File: server/ingestion.py
import os
import glob

def load_feature_files(path):
    files = glob.glob(os.path.join(path, "**/*.feature"), recursive=True)
    return [(f, open(f, "r").read()) for f in files]

def load_step_definitions(path):
    files = glob.glob(os.path.join(path, "**/*.js"), recursive=True)
    return [(f, open(f, "r").read()) for f in files]

File: src/ingestion.js
import fs from "fs";
import path from "path";
import glob from "glob";

export function loadFeatureFiles(dir) {
  const files = glob.sync(path.join(dir, "**/*.feature"));
  return files.map(f => ({ id: f, text: fs.readFileSync(f, "utf8") }));
}

export function loadStepDefinitions(dir) {
  const files = glob.sync(path.join(dir, "**/*.js"));
  return files.map(f => ({ id: f, text: fs.readFileSync(f, "utf8") }));
}
